### PR TITLE
Add transport.rest to Transportation

### PR DESCRIPTION
**Pull request by konhi**
- [x] My submission is formatted according to the guidelines in the [contributing guide](/CONTRIBUTING.md)
  - I didn't follow the rule about mentioning TLD, as it's inseparable in this situation
- [x] My addition is ordered alphabetically
- [x] My submission has a useful description
- [x] The description does not have more than 100 characters
- [x] The description does not end with punctuation
- [x] Each table column is padded with one space on either side
- [x] I have searched the repository for any relevant issues or pull requests
- [x] Any category I am creating has the minimum requirement of 3 items
- [x] All changes have been [squashed][squash-link] into a single commit

cc @derhuerst

[squash-link]: <https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1664,6 +1664,7 @@ API | Description | Auth | HTTPS | CORS |
 | [Transport for Vancouver, Canada](https://developer.translink.ca/) | TransLink | `OAuth` | Yes | Unknown |
 | [Transport for Washington, US](https://developer.wmata.com/) | Washington Metro transport API | `OAuth` | Yes | Unknown |
 | [Tripadvisor](https://developer-tripadvisor.com/home/) | Rating content for a hotel, restaurant, attraction or destination | `apiKey` | Yes | Unknown |
+| [transport.rest](https://transport.rest) | Community maintained, developer-friendly public transport API | No | Yes | Yes |
```
Comment (derhuerst) [inline README.md:1667]: You may want to add that it currently focuses on Germany, although that might change in the future.

Comment (konhi) [reply README.md:1667]: Thank you for the suggestion!
I currently have an interest in Polish public transport, and I'm actively working on [poland-public-transport-api](https://github.com/konhi/poland-public-transport-api/), so I think that change isn't required!

Comment (derhuerst) [reply README.md:1667]: Do you know @MKuranowski and their [list of Polish GTFS feeds](https://mkuran.pl/gtfs/)? They seem very involved with everything related to open public transport data in Poland!

Comment (konhi) [reply README.md:1667]: Wow, I did not know! That's a big help, thanks so much. I'm really, really impressed of your knowledge about all things related to public transport.

**Review by matheusfelipeog — APPROVED** [review]
Thanks for your contribution @konhi <3

**Pull request merged**